src/App.tsx

```diff
@@ -1,26 +1,162 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { USER_ID, createTodo, getTodos, removeTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './TodoList';
+import { Footer } from './Footer';
+import { Errors } from './types/Errors';
+import { FilterBy } from './types/FiilterBy';
+import { ErrorNotification } from './ErrorNotification';
+
+type FilterTheTodos = (todos: Todo[], filterBy: FilterBy) => Todo[];
+
+const getFilteredTodos: FilterTheTodos = (todos, filterBy) => {
+  let filteredTodos = todos;
 
-const USER_ID = 0;
+  if (filterBy !== FilterBy.All) {
+    filteredTodos = filteredTodos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.Active:
+          return !todo.completed;
+        case FilterBy.Completed:
+          return todo.completed;
+        default:
+          throw new Error('Unknown filter type');
+      }
```

Comment:
You have a filter type - it is a bad practice to throw here in such way, better will be to implement such case in error type.